rest/model/perf.go

```diff
@@ -0,0 +1,200 @@
+package model
+
+import (
+	dbmodel "github.com/evergreen-ci/sink/model"
+	"github.com/pkg/errors"
+)
+
+type APIPerformanceResult struct {
+	Name         APIString              `json:"name"`
+	Info         APIPerformanceResultID `json:"info"`
+	CreatedAt    APITime                `json:"create_at"`
+	CompletedAt  APITime                `json:"completed_at"`
+	Version      int                    `json:"version"`
+	Source       []APIArtifactInfo      `json:"source_info"`
+	AuxilaryData []APIArtifactInfo      `json:aux_data"`
+	Total        *APIPerformancePoint   `json:"total"`
+	Rollups      *APIPerfRollups        `json:"rollups"`
+}
+
+func (apiResult *APIPerformanceResult) Import(i interface{}) error {
+	switch r := i.(type) {
+	case dbmodel.PerformanceResult:
+		apiResult.Name = ToAPIString(r.ID)
+		apiResult.CreatedAt = NewTime(r.CreatedAt)
+		apiResult.CompletedAt = NewTime(r.CompletedAt)
+		apiResult.Version = r.Version
+		apiResult.Info = getPerformanceResultID(r.Info)
+
+		total := getPerformancePoint(r.Total)
+		apiResult.Total = &total
+		rollups := getPerfRollups(r.Rollups)
+		apiResult.Rollups = &rollups
+
+		var apiSource []APIArtifactInfo
+		for _, artifactInfo := range r.Source {
+			apiArtifactInfo := getArtifactInfo(artifactInfo)
+			apiSource = append(apiSource, apiArtifactInfo)
+		}
+		apiResult.Source = apiSource
+
+		var apiAuxilaryData []APIArtifactInfo
+		for _, artifactInfo := range r.AuxilaryData {
+			apiArtifactInfo := getArtifactInfo(artifactInfo)
+			apiAuxilaryData = append(apiAuxilaryData, apiArtifactInfo)
+		}
+		apiResult.AuxilaryData = apiAuxilaryData
+	default:
+		return errors.New("incorrect type when fetching converting PerformanceResult type")
+	}
+	return nil
+}
+
+func (apiResult *APIPerformanceResult) Export(i interface{}) (interface{}, error) {
+	return nil, errors.Errorf("Export is not implemented for APIPerformanceResult")
+}
+
+type APIPerformanceResultID struct {
+	Project   APIString        `json:"project"`
+	Version   APIString        `json:"version"`
+	TaskName  APIString        `json:"task_name"`
+	TaskID    APIString        `json:"task_id"`
+	Execution int              `json:"execution"`
+	TestName  APIString        `json:"test_name"`
+	Trial     int              `json:"trial"`
+	Parent    APIString        `json:"parent"`
```

Comment:
Shouldn't this be a pointer to another struct? 
From the [supplemental design doc|https://docs.google.com/document/d/1Bi-278IaeAwc5RCFal8Ww_lKMWUGVeDRLZhG8oYqJTM/edit#] you wrote:
{
    "project": <evergreen project>,
    "version": <evergreen version id>,
    "task_id": <evergreen task id>,
    "task_name": <task display name>,
    "execution": <int>,     // evergreen execution id
    "test_name": <string>,  // arbitrary, describes a unique experiment/value
    "trial": <int>,         // repeated execution
    "parent": <infoHash>,   // pointer to parent test
    "tags": [...],          // arbitrary test-supplied metadata
    "args" {},              // map of strings to ints as test parameters
    "events": [...],        // 0 or more Events
}

Reply:
I think this is correct as written.

Reply:
Thanks